### PR TITLE
Set static parameters during manifests load instead of updating it after

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Signed-off-by: Artyom Lukianov <alukiano@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
there is a significant overlap with previous https://github.com/k8stopologyawareschedwg/deployer/pull/60
let's see how we can merge the two PRs.

**Comment by cynepco3hahue** [comment]
> there is a significant overlap with previous #60
> let's see how we can merge the two PRs.

I do not have any problem first merging yours, just let not provide additional infra around arguments, because we should set arguments only once, at least now it is how it looks.

**Comment by fromanirh** [comment]
> > there is a significant overlap with previous #60
> > let's see how we can merge the two PRs.
> 
> I do not have any problem first merging yours, just let not provide additional infra around arguments, because we should set arguments only once, at least now it is how it looks.

I agree my PR does much more (maybe too much for a single PR). So Let's move forward with yours, and I'll later rebase (/split) my PR on top. So I'll review and send the minimum amount of comments to move forward and merge.

**Comment by cynepco3hahue** [comment]
> I agree my PR does much more (maybe too much for a single PR). So Let's move forward with yours, and I'll later rebase (/split) my PR on top. So I'll review and send the minimum amount of comments to move forward and merge.

Great, thanks!

File: pkg/commands/render.go [hunk 1]
```diff
@@ -129,14 +129,16 @@ func makeRTEObjects(commonOpts *CommonOptions) ([]client.Object, string, error)
...
+	mf, err := rtemanifests.GetManifests(commonOpts.UserPlatform, namespace)
 	if err != nil {
 		return nil, namespace, err
 	}
 	mf = mf.Update(rtemanifests.UpdateOptions{
-		ConfigData:       commonOpts.RTEConfigData,
-		PullIfNotPresent: commonOpts.PullIfNotPresent,
-		Namespace:        namespace,
+		ConfigData: commonOpts.RTEConfigData,
+		DaemonSetOptions: &manifests.DaemonSetUpdateOptions{
```
Comment (fromanirh) [inline pkg/commands/render.go:138]: I'm really not sold regarding the need of a substruct. Maybe we can move back `PullIfNotPresent` in the `rtemanifests.UpdateOptions`.

File: pkg/manifests/manifests.go [hunk 2]
```diff
@@ -259,6 +288,42 @@ func DaemonSet(component string) (*appsv1.DaemonSet, error) {
 	if !ok {
 		return nil, fmt.Errorf("unexpected type, got %t", obj)
 	}
+
+	for i := range ds.Spec.Template.Spec.Containers {
+		c := &ds.Spec.Template.Spec.Containers[i]
+		if c.Name == containerNameRTE {
```
Comment (fromanirh) [inline pkg/manifests/manifests.go:294]: the other way around please:
```
if c.Name == containerNameRTE {
  continue
}
```

File: pkg/manifests/manifests.go [hunk 3]
```diff
@@ -259,6 +288,42 @@ func DaemonSet(component string) (*appsv1.DaemonSet, error) {
...
+			c.Image = images.ResourceTopologyExporterImage
+			c.Command = []string{
```
Comment (fromanirh) [inline pkg/manifests/manifests.go:296]: this is better than before but I'm not really sold about it. But we can surely try and evolve incrementally.

**Comment by fromanirh** [comment]
TL;DR: I'd avoid the `DaemonSetUpdateOptions` unless it's really really needed (let's do a deep cleanup later, since this is not going to be the final cleanup) and please let's simplify a bit the  update logic in https://github.com/k8stopologyawareschedwg/deployer/pull/64/files/0c9c92634445b37f0f8e6a333e0fe447fa00dec6#diff-1f9d2d18d4cabd8262e585b932330183f7cb9b544ac9f27c8bcb3ef9c8d14376 by swapping the guard. Everything else is good enough and we will improve incrementally.

File: pkg/commands/render.go (hunk 1 above)
Comment (cynepco3hahue) [reply pkg/commands/render.go:138]: yee, I thought I will need additional parameters. Will revert it

File: pkg/manifests/manifests.go (hunk 3 above)
Comment (fromanirh) [reply pkg/manifests/manifests.go:296]: So let's go with this, we will improve later if needed once we have a clearer picture

**Comment by fromanirh** [comment]
/lgtm
good enough simplification, we can incrementally improve later
